lib/credentials/VaultServiceCredentials.js

```diff
@@ -0,0 +1,135 @@
+const http = require('http');
+const joi = require('@hapi/joi');
+const { Credentials, STS } = require('aws-sdk');
+
+const errors = require('arsenal').errors;
+const { Logger } = require('werelogs');
+const AccountCredentials = require('./AccountCredentials');
+
+const configJoi = {
+    stsclient: joi.object().required(),
+    extension: joi.string().required(),
+    roleArn: joi.string().required(),
+    refreshCredsAnticipationSeconds: joi.number().greater(0).default(60),
+};
+
+/**
+* Manages and refreshes credentials as needed through assuming a role.
+*
+* This class extends AWS' Credentials class and implements the refresh
+* method to refresh credentials once they are expired.
+*/
+class VaultServiceCredentials extends Credentials {
+    /**
+     * constructor
+     * @param {object} stsclient - stsclient instance
+     * @param {string} extension - name of the extension
+     * @param {string} roleArn - ARN of the role
+     * @param {RequestLogger} log - request logger instance
+     * @param {number} refreshCredsAnticipationSeconds - credentials
+     * must be refreshed earlier than their effective expiration time
+     * to avoid race conditions, this sets from how many seconds
+     * before expiration time credentials are refreshed
+     */
+    constructor(stsclient, extension, roleArn, log,
+                refreshCredsAnticipationSeconds) {
+        super();
+        const params = joi.attempt({
+            stsclient,
+            extension,
+            roleArn,
+            refreshCredsAnticipationSeconds,
+        }, configJoi);
+
+        this._stsclient = stsclient;
+        this._log = new Logger('Backbeat').newRequestLogger(log.getUids());
+        this._extension = extension;
+        this._roleArn = roleArn;
+        this._refreshCredsAnticipationSeconds = params.refreshCredsAnticipationSeconds;
```

Comment:
```suggestion
        this._refreshCredsAnticipationSeconds = refreshCredsAnticipationSeconds;
```
Already destructured the var

Reply:
`params.refreshCredsAnticipation` is preferred since it would have a default set by the `joi` validation

Reply:
oh nevermind yea, I see how it works now.